dask/bag/core.py

```diff
@@ -244,19 +246,34 @@ def __init__(self, dsk, name, npartitions):
         self.npartitions = npartitions
         self.str = StringAccessor(self)
 
-    def map(self, func):
+    def map(self, func, *args):
         """ Map a function across all elements in collection
 
         >>> import dask.bag as db
         >>> b = db.from_sequence(range(5))
         >>> list(b.map(lambda x: x * 10))  # doctest: +SKIP
         [0, 10, 20, 30, 40]
+
+        Any additional arguments get passed to the function _before_ the data
+        argument; argument values may either by concrete or a dask computation.
```

Comment:
Yeah, I also find after to be more intuitive; I think it would take defining a custom function like `map_with_args(func, args, iterable)` to use in place of map when we have any arguments, since it's not possible to partially apply from the end of argument list.  I'll update with this variant to make the example concrete and see if it works :-)